### ISSUE TITLE
Loading a model from local files achieves way too lower accuracy in comparison to model downloading

### ISSUE THREAD

**Issue by sstojanoska** [huggingface/transformers#10177]
## Environment info

- `transformers` version: 4.3.2
- Platform: Linux-4.19.112+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.6.9
- PyTorch version (GPU?): 1.7.0+cu101 (True)
- Tensorflow version (GPU?): 2.4.1 (True)
- Using GPU in script?: yes
- Using distributed or parallel set-up in script?: no

### Who can help

@LysandreJik, @sgugger, @patrickvonplaten

## Information

I am using a few models from the 'Models hub' ( https://huggingface.co/models ) and
I am working on token classification task. Since I have to repeat some of the experiments several times I have downloaded  the models locally, to avoid downloading on each run.

Downloading:
```
model_name = "X"
model = AutoModelForTokenClassification.from_pretrained(model_name, num_labels=len(tag2idx))
tokenizer = AutoTokenizer.from_pretrained(model_name)
```
Saving locally the model and tokenizer ( and renaming tokenizer_config.json to config.json)
```
model.save_pretrained("/content/drive/MyDrive/model/model_name/")
tokenizer.save_pretrained("/content/drive/MyDrive/tokenizer/model_name/")
```
Loading the model and tokenizer from local directories
```
config = AutoConfig.from_pretrained("/content/drive/MyDrive/model/model_name/")
config.num_labels = len(tag2idx)
model = AutoModelForTokenClassification.from_config(config)
tokenizer = AutoTokenizer.from_pretrained("/content/drive/MyDrive/tokenizer/model_name/")
```

The problem arises when using the model from local files. I have noticed that the accuracy of the local model for the exact same configuration (same data, number of epochs, lr,  etc.) is around 20% lower than downloading the model for each experiment.
I would like to know why this happens.

## Expected behavior
The model should achieve similar results whether being loaded from local files or downloaded from the models hub.

**Comment by sgugger** [comment]
> The problem arises when using the model from local files. I have noticed that the accuracy of the local model for the exact same configuration (same data, number of epochs, lr, etc.) is around 20% lower than downloading the model for each experiment.

This is extremely vague and we can't help you solve your bug if you don't give us something more tangible than this. What code are you then running? What's the difference in accuracy?

**Comment by sstojanoska** [comment]
Thanks. It was my mistake. 
The issue was that I was trying to dynamically change the model config file and reuse it for models with different number of output labels.
